test/index.js

```diff
@@ -627,8 +628,9 @@ describe('def', function() {
 
     throws(function() { f('XXX'); },
            errorEq(TypeError,
-                   '‘f’ is expected to return a value of type ' +
-                   '(Nullable String); returned 42'));
+                   '‘f’ expected its first argument and return value to be ' +
+                   'of the same type; the first argument is of type String' +
+                   ', however the return value, 42, is of type Number'));
```

Comment:
We appear to be comparing the `a`s only. The first argument is really of type `Nullable String` in this context.